platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ExportMigrationEntryImpl.java

```diff
@@ -0,0 +1,318 @@
+        if (stored == null) {
+            this.stored = false; // until proven otherwise
+            recordEntry();
+            try (final OutputStream os = getOutputStream()) {
+                this.stored = getReport().wasIOSuccessful(() -> consumer.accept(getReport(), os));
+            } catch (ExportIOException e) { // special case indicating the I/O error occurred while writing to the zip which would invalidate the zip so we are forced to abort
+                throw newError("failed to be exported", e.getCause());
+            } catch (IOException e) { // here it means the error came out of reading/processing the input file/stream where it is safe to continue with the next entry, so don't abort
+                getReport().record(newError("failed to be exported", e));
+            } catch (MigrationException e) {
+                throw e;
+            }
+        }
+        return stored;
+    }
+
+    @Override
+    public Optional<ExportMigrationEntry> getPropertyReferencedEntry(String name,
+            BiPredicate<MigrationReport, String> validator) {
+        Validate.notNull(name, "invalid null java property name");
+        Validate.notNull(validator, "invalid null validator");
+        final ExportMigrationJavaPropertyReferencedEntryImpl me = properties.get(name);
+
+        if (me != null) {
+            return Optional.of(me);
+        }
+        try {
+            final String val = getJavaPropertyValue(name);
+
+            if (!validator.test(getReport(), val)) {
+                return Optional.empty();
+            } else if (val == null) {
+                getReport().record(new MigrationException(Messages.EXPORT_JAVA_PROPERTY_NOT_DEFINED_ERROR,
+                        name,
+                        path));
+                return Optional.empty();
+            } else if (val.isEmpty()) {
+                getReport().record(new MigrationException(Messages.EXPORT_JAVA_PROPERTY_IS_EMPTY_ERROR,
+                        name,
+                        path));
+                return Optional.empty();
+            }
+            final ExportMigrationJavaPropertyReferencedEntryImpl prop =
+                    new ExportMigrationJavaPropertyReferencedEntryImpl(context, path, name, val);
+
+            properties.put(name, prop);
+            return Optional.of(prop);
+        } catch (IOException e) {
+            getReport().record(new MigrationException(Messages.EXPORT_JAVA_PROPERTY_LOAD_ERROR,
+                    name,
+                    path,
+                    e));
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    protected ExportMigrationContextImpl getContext() {
+        return context;
+    }
+
+    protected Path getAbsolutePath() {
+        return absolutePath;
+    }
+
+    protected File getFile() {
+        return file;
+    }
+
+    /**
+     * Called to record that this entry is being processed.
+     */
+    protected void recordEntry() { // nothing to record here
```

Comment:
I see why you did this, but it still bothers me 😉

Reply:
My preference would have been to let the report check the type of entry and decided how to record it but after discussion with Eric, we opted for this approach.